### ISSUE TITLE
ETA on switching to Bootstrap 4?

### ISSUE THREAD

**Issue by AlejandroFlorin** [MarkPieszak/aspnetcore-angular2-universal#475]
Is there an ETA to switch to Bootstrap 4?

Is the plan to wait for Bootstrap 4 leave beta?

Thanks,

Alex

**Comment by isaac2004** [comment]
I would assume so, plus will probably need to wait for ngx-bootstrap as well

**Comment by alexbenitez** [comment]
Would it be possible to add a section in the readme (or in this issue) detailing the steps to upgrade the starter to Bootstrap 4 beta 2?

**Comment by isaac2004** [comment]
Https://GitHub.com/isaac2004/LevinBlog is a fork of this repo ( before ng-5) and is using BS4

**Comment by alexbenitez** [comment]
Thx! I'll take a look.

**Comment by isaac2004** [comment]
@alexbenitez you get it straightened out?

**Comment by alexbenitez** [comment]
Thanks @isaac2004  for asking! Unfortunately I'm still missing something. I went back to basis and found this basic sample that only has bootstrap 3 and ngx-bootstrap (https://github.com/loiane/angular-bootstrap-example). The sample works perfect but when I tried upgrading it to bootstrap 4 (e.g. npm install bootstrap@next ngx-bootstrap@next --save ) got the same kind of problems with navbars that I experienced with my code based on this starter.  Basically the dropdowns don't work and/or I see hyperlinks instead.

These are the versions that I'm trying currently. I noticed you repository is using the alpha version but I assume it doesn't matter?

    "bootstrap": "^4.0.0-beta.2",
    "ngx-bootstrap": "^2.0.0-beta.7",

Other than upgrading with npm are there other things that need to be done for bs4 beta.2 to work? (I tried it with several ngx-bootstrap versions).

/Alex

**Comment by isaac2004** [comment]
Did you check to make sure the classes didn't change? I know I had a few scenarios where a class name changed with the upgrade.

**Comment by alexbenitez** [comment]
Yes I did.. Unless I missed something, I don't see class name or directive changes however the css styles for the classes are different (which is expected). By inspecting the elements with dev tools I see that with bs3 there is a  "display:block" applied whereas with bs4 it uses "display:flex" (using the container-fluid class). This pushes the menus all the way to the right (bs4). The list items are not displayed inline in bs4, and displayed raw, etc... For the time being I'll remain with bs3 (or try later with ng-bootstrap if they do deal with navbar - I don't have an immediate need to support bs3 - ). I just can't find enough information to know if this is a bootstrap outstanding issue or something that needs to be done differently by design. The repo that I referenced above is a good sample to reproduce the problem (by just npm installing bs4 and ngx). This is more an issue that has to do with ngx-boostrap / bootstrap not really specific to the starter (... but still for this repo  it would be great documenting the steps to upgrade it to bs4...  if there is something else other than just run npm install with the recent versions).

**Comment by GetTaxSolutions** [comment]
We're also want to change the bootstrap version to 4.0.0 beta 2 and have questions how to do it?
1. I didn't find package 'bootstrap-sass' for this version. Is not available and we can remove it from project for now. Right? So I commented this in app.component.scss:
`/* Import Bootstrap & Fonts */
/*$icon-font-path: '~bootstrap-sass/assets/fonts/bootstrap/';
@import "~bootstrap-sass/assets/stylesheets/bootstrap";*/`
2. I followed instruction from bootstrap how to install on Webpack - [https://getbootstrap.com/docs/4.0/getting-started/webpack/](url)
So I have to add styles in javascripts in our project.
2.1. Add styles using bootstrap compiled version of CSS: In webpack.config.vendor.js added the styles in nonTreeShakableModules:
`const nonTreeShakableModules = [
......
    'bootstrap/dist/css/bootstrap.min.css',
...]`
Is it enough to load all bootstrap styles?
2.2. Add bootstrap styles in project. Add bootstrap module in nonTreeShakableModules
`const nonTreeShakableModules = [
......
    'bootstrap',
...]`
And register it and related libraries in ProvidePlugin:
`new webpack.ProvidePlugin({
            $: 'jquery',
            jQuery: 'jquery',
            'window.jQuery': 'jquery',
            Popper: ['popper.js', 'default']
          })`
So it have to load the bootstrap scripts. Right?

I expected to see all bootstrap functionality but actually the styles weren't loaded. Not sure for the scripts. Why is that? Shouldn't Webpack load and apply all entry points including bootstrap.min.css styles?

I tryied to import them again in app.component.scss:
`@import '~bootstrap/dist/css/bootstrap.min.css';`
After that the part of styles was applied but the home page had some misalignments and navigation menu is missed from the page. Can you helm with some details instructions how to switch to this bootstrap version properly. Where is my mistake here?


Thank you

**Comment by GRIMMR3AP3R** [comment]
Done and done: 352fab8
